internal/pkg/cli/validate.go

```diff
@@ -134,3 +158,37 @@ func stringPortValidation(val string) error {
 	}
 	return nil
 }
+
+// s3 bucket names: 'a-zA-Z0-9.-'
+func s3BucketNameValidation(val interface{}) error {
+	s, ok := val.(string)
+	if !ok {
+		return errValueNotAString
+	}
+	if len(s) < 3 || len(s) > 63 {
+		return errS3ValueBadSize
+	}
+	for _, r := range s {
+		if !(unicode.IsDigit(r) || unicode.IsLetter(r) || r == '.' || r == '-') {
+			return errS3ValueBadFormat
+		}
+	}
+	return nil
+}
+
+// Dynameo table names: 'a-zA-Z0-9.-_'
+func dynamoTableNameValidation(val interface{}) error {
+	s, ok := val.(string)
+	if !ok {
+		return errValueNotAString
+	}
+	if len(s) < 3 || len(s) > 255 {
+		return errDDBValueBadSize
+	}
+	for _, r := range s {
+		if !(unicode.IsDigit(r) || unicode.IsLetter(r) || r == '.' || r == '-' || r == '_') {
```

Comment:
Is there a reason why we don't want to use a regex matcher instead? 
We could move that to its own function and then use that validator if a storage type is not matched, what do you think?

Reply:
Do you mean to just add separate "basic alphanumeric matching" regex function that's separate from the ddb matching?